applications/convection_diffusion_application/python_scripts/convection_diffusion_base_solver.py

```diff
@@ -0,0 +1,757 @@
+                "reaction_variable"             : "REACTION_FLUX"
+            },
+            "time_stepping" : { },
+            "reform_dofs_at_each_step": false,
+            "line_search": false,
+            "compute_reactions": true,
+            "block_builder": true,
+            "clear_storage": false,
+            "move_mesh_flag": false,
+            "convergence_criterion": "residual_criterion",
+            "solution_relative_tolerance": 1.0e-4,
+            "solution_absolute_tolerance": 1.0e-9,
+            "residual_relative_tolerance": 1.0e-4,
+            "residual_absolute_tolerance": 1.0e-9,
+            "max_iteration": 10,
+            "linear_solver_settings":{
+                "solver_type": "BICGSTABSolver",
+                "preconditioner_type": "DiagonalPreconditioner",
+                "max_iteration": 5000,
+                "tolerance": 1e-9,
+                "scaling": false
+            },
+            "element_replace_settings" : {
+                "element_name" : "EulerianConvDiff",
+                "condition_name" : "Condition"
+            },
+            "problem_domain_sub_model_part_list": ["conv_diff_body"],
+            "processes_sub_model_part_list": [""],
+            "auxiliary_variables_list" : []
+        }
+        """)
+
+        # Adding warnings
+        if not custom_settings.Has("convection_diffusion_variables"):
+            self.print_warning_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "W-A-R-N-I-N-G: CONVECTION DIFFUSION  VARIABLES NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"])
+        else:
+            if not custom_settings["convection_diffusion_variables"].Has("density_variable"):
+                self.print_warning_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "W-A-R-N-I-N-G: DENSITY VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["density_variable"].GetString())
+            if not custom_settings["convection_diffusion_variables"].Has("diffusion_variable"):
+                self.print_warning_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "W-A-R-N-I-N-G: DIFUSSION VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["diffusion_variable"].GetString())
+            if not custom_settings["convection_diffusion_variables"].Has("unknown_variable"):
+                self.print_warning_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "W-A-R-N-I-N-G: UNKNOWN VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["unknown_variable"].GetString())
+            if not custom_settings["convection_diffusion_variables"].Has("volume_source_variable"):
+                self.print_warning_on_rank_zero("::[ConvectionDiffusionBaseSolver]:: ", "W-A-R-N-I-N-G: VOLUME SOURCE VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["volume_source_variable"].GetString())
```

Comment:
are u sure that a default is defined? for example what would it be in this case?

Reply:
The ones in the default parameters